stream/src/main/scala/org/apache/pekko/stream/impl/JsonObjectParser.scala

```diff
@@ -117,21 +117,22 @@ import pekko.util.ByteString
     val bufSize = buffer.length
 
     skipToNextObject(bufSize)
+    val maxObjectLengthIndex = if (pos + maximumObjectLength < 0) Int.MaxValue else pos + maximumObjectLength
```

Comment:
@He-Pin what do you think about this solution? This way we catch the overflow directly. On the other hand, such a big `maximumObjectLength` is unlikely to work anyway because we cannot accommodate such a big object including any padding into our buffer... (i.e. we'll still fail in `offer`). Supporting huge objects is not economical in any case right now because of the buffer management issue.